src/features/swap/helpers/__tests__/utils.test.ts

```diff
@@ -175,4 +181,62 @@ describe('Swap helpers', () => {
       expect(result).toEqual(50)
     })
   })
+
+  describe('isOrderPartiallyFilled', () => {
+    it('returns true if a buy order is partially filled', () => {
+      const mockOrder = {
+        executedBuyAmount: '10',
+        buyAmount: '100000000000000000000', // 100 tokens
+        executedSellAmount: '50000000000000000000', // 50 tokens
+        sellAmount: '100000000000000000000', // 100 tokens
+        kind: 'buy',
+      } as unknown as SwapOrder
+
+      const result = isOrderPartiallyFilled(mockOrder)
+
+      expect(result).toBe(true)
+    })
+
+    it('returns false if a buy order is fully filled', () => {
+      const mockOrder = {
+        executedBuyAmount: '0',
+        buyAmount: '100000000000000000000', // 100 tokens
+        executedSellAmount: '100000000000000000000', // 100 tokens
+        sellAmount: '100000000000000000000', // 100 tokens
+        kind: 'buy',
+      } as unknown as SwapOrder
+
+      const result = isOrderPartiallyFilled(mockOrder)
+
+      expect(result).toBe(false)
+    })
+
+    it('returns true if a sell order is partially filled', () => {
+      const mockOrder = {
+        sellAmount: '100000000000000000000',
+        executedSellAmount: '10',
+        executedBuyAmount: '50000000000000000000', // 50 tokens
+        buyAmount: '100000000000000000000', // 100 tokens
+        kind: 'sell',
+      } as unknown as SwapOrder
+
+      const result = isOrderPartiallyFilled(mockOrder)
+
+      expect(result).toBe(true)
+    })
+
+    it('returns false if a sell order is fully filled', () => {
+      const mockOrder = {
+        sellAmount: '100000000000000000000',
+        executedSellAmount: '0',
```

Comment:
Wouldn't this order be not filled at all?

Reply:
test title is wrong. The condition for a partial fill is

executedBuyAmount !== 0 and executedBuyAmount < buyAmount.

With the test as is, I've just tested the left side of the check. 🤦